zipline/data/adjustments.py

```diff
@@ -0,0 +1,599 @@
+from collections import namedtuple
+from errno import ENOENT
+from os import remove
+
+from logbook import Logger
+import numpy as np
+from numpy import integer as any_integer
+import pandas as pd
+from pandas import Timestamp
+import six
+import sqlite3
+
+from zipline.data.bar_reader import NoDataOnDate
+from zipline.utils.input_validation import preprocess
+from zipline.utils.numpy_utils import (
+    float64_dtype,
+    int64_dtype,
+    uint32_dtype,
+)
+from zipline.utils.sqlite_utils import group_into_chunks, coerce_string_to_conn
+from ._adjustments import load_adjustments_from_sqlite
+
+log = Logger(__name__)
+
+
+SQLITE_ADJUSTMENT_TABLENAMES = frozenset(['splits', 'dividends', 'mergers'])
+
+UNPAID_QUERY_TEMPLATE = """
+SELECT sid, amount, pay_date from dividend_payouts
+WHERE ex_date=? AND sid IN ({0})
```

Comment:
why is this a string template and not a query param?

Reply:
I think it's because we use pandas.read_sql_query to execute this, which doesn't support query params, but I'm not sure. This was here previously.